agent/controller/ssm.go

```diff
@@ -0,0 +1,107 @@
+	pbclient "github.com/hoophq/hoop/common/proto/client"
+)
+
+func (a *Agent) processSSMProtocol(pkt *pb.Packet) {
+	sessionID := string(pkt.Spec[pb.SpecGatewaySessionID])
+	streamClient := pb.NewStreamWriter(a.client, pbclient.SSMConnectionWrite, pkt.Spec)
+	connParams := a.connectionParams(sessionID)
+	if connParams == nil {
+		log.Errorf("session=%s - connection params not found", sessionID)
+		a.sendClientSessionClose(sessionID, "connection params not found, contact the administrator")
+		return
+	}
+
+	clientConnectionID := string(pkt.Spec[pb.SpecClientConnectionID])
+	if clientConnectionID == "" {
+		log.Println("connection id not found in memory")
+		a.sendClientSessionClose(sessionID, "connection id not found, contact the administrator")
+		return
+	}
+
+	clientConnectionIDKey := fmt.Sprintf("%s:%s", sessionID, string(clientConnectionID))
+	clientObj := a.connStore.Get(clientConnectionIDKey)
+	if serverWriter, ok := clientObj.(io.WriteCloser); ok {
+		if _, err := serverWriter.Write(pkt.Payload); err != nil {
+			log.Errorf("failed sending packet, err=%v", err)
+			a.sendClientSessionClose(sessionID, "fail to write packet")
+			_ = serverWriter.Close()
+		}
+		return
+	}
+
+	targetInstance := string(pkt.Spec[pb.SpecInstanceId])
+	if targetInstance == "" {
+		log.Println("missing aws instance id")
+		a.sendClientSessionClose(sessionID, "missing aws instance id, contact the administrator")
+		return
+	}
+
+	connenv, err := parseConnectionEnvVars(connParams.EnvVars, pb.ConnectionTypeSSM)
+	if err != nil {
+		log.Error("postgres credentials not found in memory, err=%v", err)
+		a.sendClientSessionClose(sessionID, "credentials are empty, contact the administrator")
+		return
+	}
+
+	log.Infof("session=%v - starting postgres connection at %v:%v", sessionID, connenv.host, connenv.port)
+
+	var dataMaskingEntityTypesData string
+	if connParams.DataMaskingEntityTypesData != nil {
+		dataMaskingEntityTypesData = string(connParams.DataMaskingEntityTypesData)
+	}
+	var guardRailRules string
+	if connParams.GuardRailRules != nil {
+		guardRailRules = string(connParams.GuardRailRules)
+	}
+
+	var initPacket service.OpenDataChannelInput
+	if err := json.Unmarshal(pkt.Payload, &initPacket); err != nil {
+		errMsg := fmt.Sprintf("failed connecting with ssm server, err=%v", err)
+		log.Errorf(errMsg)
+		a.sendClientSessionClose(sessionID, errMsg)
+		return
+	}
+
+	opts := map[string]string{
+		"sid":                       sessionID,
+		"dlp_provider":              connParams.DlpProvider,
+		"dlp_mode":                  connParams.DlpMode,
+		"mspresidio_analyzer_url":   connParams.DlpPresidioAnalyzerURL,
+		"mspresidio_anonymizer_url": connParams.DlpPresidioAnonymizerURL,
+		"dlp_gcp_credentials":       connParams.DlpGcpRawCredentialsJSON,
+		"dlp_info_types":            strings.Join(connParams.DLPInfoTypes, ","),
+		"dlp_masking_character":     "#",
+		"data_masking_entity_data":  dataMaskingEntityTypesData,
+		"guard_rail_rules":          guardRailRules,
```

Comment:
Remove everything from here, we're not using right now. You should pass it after we starting implementing these feature.